src/radical/entk/execman/rp/task_processor.py

```diff
@@ -518,8 +518,8 @@ def create_task_from_cu(cu, prof=None):
         task.parent_pipeline['name'] = cu.name.split(',')[5].strip()
         task.rts_uid                 = cu.uid
 
-        if cu.state == rp.DONE: task.exit_code = 0
-        else                  : task.exit_code = 1
+        if cu.state == rp.DONE                    : task.exit_code = 0
+        elif cu.state in [rp.FAILED, rp.CANCELED] : task.exit_code = 1
```

Comment:
`exit_code` should remain `None` if the task did not go through execution for some reason, so I think the `else` is indeed not needed.  But I'd argue it also should remain `None` on `CANCELED` -- the exit code could not be determined in that case.

Reply:
EnTK uses the exit code to decide if a task has finished correctly or not. It may have to be changed in the future. But I think it should not right now.